src/time.rs

```diff
@@ -0,0 +1,58 @@
+//! Time-Related Functions and Types
+
+pub use chrono::{DateTime, NaiveDateTime, TimeZone, Utc};
+pub use prost_types::Timestamp;
+
+/// Converts a prost [`Timestamp`] value to a chrono [`DateTime<Utc>`] value.
+/// Will return [`None`] if [`Timestamp`] `nanos` value is negative
+pub fn timestamp_to_datetime(ts: &Timestamp) -> Option<DateTime<Utc>> {
+    let nanos: u32 = ts.nanos.try_into().ok()?;
+    let ndt: NaiveDateTime = NaiveDateTime::from_timestamp_opt(ts.seconds, nanos)?;
+    Some(DateTime::<Utc>::from_utc(ndt, Utc))
+}
+
+/// Converts a chrono [`DateTime<Utc>`] value to a prost [`Timestamp`]() value.
+pub fn datetime_to_timestamp(dt: &DateTime<Utc>) -> Option<Timestamp> {
+    let seconds = dt.timestamp();
+
+    let nanos: i32 = dt.timestamp_subsec_nanos().try_into().ok()?;
+
```

Comment:
Shouldn't we handle the error?

Reply:
The `.ok()` will make sure `None` is returned here as well if the result would be an error.
From the `ok()` docs:
```
Converts self into an [Option<T>], consuming self, and discarding the error, if any.
```